src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -295,6 +296,21 @@ private static PackageSpec ToPackageSpec(ProjectNames projectNames, IVsProjectRe
             return packageSpec;
         }
 
+        /// <summary>
+        /// Tries to get  the projectId for a project from the project system cache using the project unique name.
+        /// </summary>
+        /// <param name="projectUniqueName">Unique name for the project.</param>
+        /// <param name="cache">ProjectSystem cache</param>
+        /// <param name="result">Contains the projectId if the return result was true else it is set to null.</param>
+        /// <returns>Returns bool indicating if the operation was successful.</returns>
+        private static bool TryGetProjectIdFromCache(string projectUniqueName, IProjectSystemCache cache, out string result)
+        {
+            result = null;
+
+            return cache.TryGetNuGetProject(projectUniqueName, out var cacheEntry) &&
```

Comment:
When nomination happens before Solution add project or initialization of NuGet solution manager for most sdk style projects, this will not return any cache entry since it'll be added as part of this nomination. So we need to cover that scenario and make sure we update the PackageSpec with the ProjectId